File: API/Entities/OrderAggregate/Order.cs
using System;

namespace API.Entities.OrderAggregate;

public class Order
{
    public int Id { get; set; }
    public required string BuyerEmail { get; set; }
    public required ShippingAddress ShippingAddress { get; set; }
    public DateTime OrderDate { get; set; } = DateTime.UtcNow;
    public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
    public long Subtotal { get; set; }
    public long DeliveryFee { get; set; }

    public long Discount { get; set; }

    public required string PaymentIntentId { get; set; }
    public long Total => Subtotal + DeliveryFee - Discount;

    public required PaymentSummary PaymentSummary { get; set; }
    public List<OrderItem> OrderItems { get; set; } = [];
}

File: client/src/app/models/order.ts
export type Order = {
    id: number;
    buyerEmail: string;
    shippingAddress: ShippingAddress;
    orderDate: string;
    orderStatus: string;
    subtotal: number;
    deliveryFee: number;
    discount: number;
    total: number;
    paymentSummary: PaymentSummary;
    orderItems: OrderItem[];
}

export type ShippingAddress = {
    name: string;
    line1: string;
    line2?: string | undefined | null;
    city: string;
    state: string;
    postal_code: string;
    country: string;
}

export type PaymentSummary = {
    last4: number | string;
    brand: string;
    exp_month: number;
    exp_year: number;
}

export type OrderItem = {
    productId: number;
    name: string;
    pictureUrl: string;
    price: number;
    quantity: number;
}

export type CreateOrder = {
    shippingAddress: ShippingAddress;
    paymentSummary: PaymentSummary;
}
